lib/service.js

```diff
@@ -0,0 +1,61 @@
+var Q = require('q');
+var _ = require('lodash');
+var assert = require('assert');
+var ComponentReader = require('./component_reader').ComponentReader;
+var compReader = new ComponentReader();
+
+var COMPONENT_PATH = process.env.COMPONENT_PATH;
+
+exports.execService = execService;
+
+var allowedMethods = {
+    verifyCredentials: verifyCredentials,
+    getMetaModel: getMetaModel,
+    selectModel: selectModel
+};
+
+function execService(method, cfg, params) {
+    assert(_.contains(_.keys(allowedMethods), method), 'Unknown service method');
+
+    function callMethod() {
+        return allowedMethods[method](cfg, params);
+    }
+
+    return compReader.init(COMPONENT_PATH).then(callMethod);
+}
+
+function verifyCredentials(cfg, params) {
+    function doVerification(verify) {
+        return Q.nfcall(verify, cfg);
+    }
+
+    function error(e) {
+        return {
+            verified: false,
+            reason: e.message
+        };
+    }
+
+    return compReader.loadVerifyCredentials()
+        .then(doVerification)
+        .catch(error);
+}
+
+function getMetaModel(cfg, params) {
+    function doGetMetaModel(module) {
+        return Q.nfcall(module.getMetaModel, cfg);
+    }
+
+    return compReader.loadTriggerOrAction(params.triggerOrAction)
+        .then(doGetMetaModel);
+}
+
+function selectModel(cfg, params) {
+    function getModel(module) {
+        assert(params.getModelMethod in module, 'Method not found');
+        return Q.nfcall(module[params.getModelMethod], cfg);
+    }
+
+    return compReader.loadTriggerOrAction(params.triggerOrAction)
```

Comment:
Looks like the only real difference between selectModel and getMetaModel functions is: `module.getMetaModel` vs `module[params.getModelMethod]`.

Am i missing something?